### PR TITLE
Add Tempo Prepackaged Server

### LINKED ISSUE #3192: Add Tempo Prepackaged Server
Add a [Tempo](https://github.com/SeldonIO/tempo) prepackaged server.

### PR DESCRIPTION
**Pull request by cliveseldon**
**What this PR does / why we need it**:

 Adds tempo prepackaged server

**Which issue(s) this PR fixes**:

Fixes #3192

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
```

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **cliveseldon**
You can assign the PR to them by writing `/assign @cliveseldon` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | fcb4b77f84936b77fc87a5f0d767c7ae3830a47a | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/integration-7j44z) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | fcb4b77f84936b77fc87a5f0d767c7ae3830a47a | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/integration-7j44z) | `/test integration`
notebooks | fcb4b77f84936b77fc87a5f0d767c7ae3830a47a | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/notebooks-g9rsm) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | fcb4b77f84936b77fc87a5f0d767c7ae3830a47a | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/integration-7j44z) | `/test integration`
notebooks | e3e7968552db01031fba2195e15d1374f6241f0b | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/notebooks-d74kt) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | e3e7968552db01031fba2195e15d1374f6241f0b | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/notebooks-d74kt) | `/test notebooks`
integration | e3e7968552db01031fba2195e15d1374f6241f0b | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/integration-k8k42) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | e3e7968552db01031fba2195e15d1374f6241f0b | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/notebooks-d74kt) | `/test notebooks`
integration | e3e7968552db01031fba2195e15d1374f6241f0b | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/integration-dv9k4) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: helm-charts/seldon-core-operator/templates/customresourcedefinition_v1_seldondeployments.machinelearning.seldon.io.yaml [hunk 1]
```diff
@@ -6,7 +6,6 @@ metadata:
   annotations:
     cert-manager.io/inject-ca-from: '{{ include "seldon.namespace" . }}/seldon-serving-cert'
     controller-gen.kubebuilder.io/version: v0.4.1
-  creationTimestamp: null
```
Comment (axsaucedo) [inline helm-charts/seldon-core-operator/templates/customresourcedefinition_v1_seldondeployments.machinelearning.seldon.io.yaml:9]: Is this due to newer version of controllergen?

File: helm-charts/seldon-core-operator/values.yaml [hunk 2]
```diff
@@ -118,7 +118,7 @@ predictor_servers:
         defaultImageVersion: "1.8.0-dev"
         image: seldonio/sklearnserver
       kfserving:
-        defaultImageVersion: "0.1.1"
+        defaultImageVersion: "0.3.1.dev7"
```
Comment (axsaucedo) [inline helm-charts/seldon-core-operator/values.yaml:121]: Should we do a release of MLServer to pin it to a stable release?

File: operator/controllers/mlserver.go [hunk 3]
```diff
@@ -35,9 +37,9 @@ func mergeMLServerContainer(existing *v1.Container, mlServer *v1.Container) *v1.
 		existing.Image = mlServer.Image
 	}
 
-	if existing.Args == nil {
-		existing.Args = mlServer.Args
-	}
+	//if existing.Args == nil {
```
Comment (axsaucedo) [inline operator/controllers/mlserver.go:40]: Is this no longer needed?

File: operator/controllers/mlserver.go [hunk 4]
```diff
@@ -197,6 +194,10 @@ func getMLServerEnvVars(pu *machinelearningv1.PredictiveUnit) ([]v1.EnvVar, erro
 			Name:  MLServerModelURIEnv,
 			Value: DefaultModelLocalMountPath,
 		},
+		{
+			Name:  MLServerTempoRuntimeEnv,
```
Comment (axsaucedo) [inline operator/controllers/mlserver.go:198]: Would it make sense to allow user to also provide overrides? or is that already covered by the podspec itself?

File: helm-charts/seldon-core-operator/templates/customresourcedefinition_v1_seldondeployments.machinelearning.seldon.io.yaml (hunk 1 above)
Comment (cliveseldon) [reply helm-charts/seldon-core-operator/templates/customresourcedefinition_v1_seldondeployments.machinelearning.seldon.io.yaml:9]: I assume so.

File: operator/controllers/mlserver.go (hunk 3 above)
Comment (cliveseldon) [reply operator/controllers/mlserver.go:40]: removed

File: operator/controllers/mlserver.go (hunk 4 above)
Comment (cliveseldon) [reply operator/controllers/mlserver.go:198]: yes, replicas could be done by sdep itself.

File: helm-charts/seldon-core-operator/values.yaml (hunk 2 above)
Comment (cliveseldon) [reply helm-charts/seldon-core-operator/values.yaml:121]: probably yes. @adriangonz

Comment (adriangonz) [reply helm-charts/seldon-core-operator/values.yaml:121]: Good point @axsaucedo.

We've just published the `0.3.2` release of MLServer (both in PyPi and Docker Hub), so we use that version tag instead.

Comment (cliveseldon) [reply helm-charts/seldon-core-operator/values.yaml:121]: will update to 0.3.2

**Comment by axsaucedo** [comment]
Integration and notebook tests pass, merging

**Pull request merged**